python/django/userdata/views.py

```diff
@@ -72,8 +77,22 @@ def post(self, request):
         serializer = self.serializer_class(data=request.DATA)
         if serializer.is_valid():
             user = serializer.object['user']
+
             token, created = Token.objects.get_or_create(user=user)
-            return Response({'token': token.key, 'user': token.user_id})
+
+            # check if first login
+            userObject = OTIUser.objects.get(username=user)
+
+             # check if this is the first time user has logged in
+            first_login = False
+            if userObject and userObject.last_login < LOGIN_FLAG_DATE:
```

Comment:
It gets set to Jan 1st, 1984 in `oti_create_user.py`.  Django 1.8 and above sensibly set the `last_login` to `None` on new user creation, but on our version of Django, `last_login` has a non-null constraint.